xcodeproj/internal/xcodeproj_macro.bzl

```diff
@@ -266,6 +283,20 @@ in your `.bazelrc` or `xcodeproj.bazelrc` file.""")
         project_name = name
     if not project_options:
         project_options = _default_project_options()
+    if not xcode_configurations:
+        xcode_configurations = {"Debug": {}}
```

Comment:
Out of the box, Xcode projects and SwiftPM packages have the ability to be built in "Debug" mode and "Release" mode. Why not mirror that here?

Reply:
Each additional configuration is a multiplier on the number of targets analyzed, serialized into the spec.json, and processed in the generator. Besides for being slower with each configuration added, for very large projects we might even run into memory limits.

Since most people survived without the release config, and because you shouldn't really be "releasing" from this generated project, I felt it was ideal not to change the default at this time.

Reply:
I guess the large projects can choose to remove the extra configuration. I just think that might be hard to debug if it's an issue, and also puts rules_xcodeproj at an unfair disadvantage when being compared with other generators that don't do this by default.

We can always change this default later, though it's harder to remove the extra config.

I just got some benchmarks for one large project, where total generation (analysis + generator) went from 43s to 80s (as expected). Maybe after we are able to spend time optimizing that more we can change the default?

Reply:
I don't feel super strongly, but do lean more on the side of having Debug/Release options enabled by default. Larger projects likely have more time to devote to tweaking and optimizing things, whereas the default should be what is a good starting point without much customization.

As far as other generators go, it's not an apples-to-apples comparison, but XcodeGen and Xcode projects auto-generated from SwiftPM packages both would create Debug/Release configurations by default and I'd lean more on feature parity than performance parity for defaults.

It's true you don't "release" from this generated project, but Swift and Objective-C code in particular often varies in behavior and certainly performance when building with or without optimizations.

Reply:
Conversation being continued in slack (sorry everyone that doesn't have access): https://bazelbuild.slack.com/archives/C0429BN0F8V/p1678378765794209?thread_ts=1677165331.047499&cid=C0429BN0F8V. I'll post back here with whatever is concluded there though.

Reply:
Deciding not to change the default for these reasons:
- The performance increase currently is a concern
- Opting out of this extra configuration is clunky
- Things with Bazel usually need to be explicit, and having this apply some automatic assumptions can actually be confusing
- By not having multiple configurations it allows for easier progressive disclosure, leading to new Bazel users to learn about configs and flags
- While it is an advanced topic, cache warming with the command-line API would by default build twice as many things (we should add a topic to `usage.md` regardless about this so people understand how to do it and what impacts its performance)

Reply:
I think that's fair, thanks for considering it and driving a discussion around it.